src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs

```diff
@@ -814,7 +814,9 @@ private void UpdateCheckboxVisibility(object sender, bool isPointerOver)
 			if (sender is ListViewItem control && control.FindDescendant<UserControl>() is UserControl userControl)
 			{
 				// Handle visual states
-				if (control.IsSelected || isPointerOver || (control.FindDescendant("SelectionCheckbox") as CheckBox)!.IsPointerOver)
+				if (UserSettingsService.FoldersSettingsService.ShowCheckboxesInDetailsView && control.IsSelected
+					|| isPointerOver || (control.FindDescendant("SelectionCheckbox") as CheckBox)!.IsPointerOver)
```

Comment:
It seems like this will show on hover even if the setting is turned off, just confirming that this is intended.